src/app/core/params.ts

```diff
@@ -0,0 +1,215 @@
+     */
+    getValue(key: string) {
+        this.__ensureHasKey(key);
+        return this.parameters[key].getValue();
+    }
+    /**
+     * Gets the paramter.
+     * @param {String} key The key is needed to search and identify the parameter
+     * @returns Returns parameter
+     * @memberof Params
+     */
+    getParameter(key: string) {
+        this.__ensureHasKey(key);
+        return this.parameters[key];
+    }
+    /**
+     * Checks if param object has unique key.
+     * @param {String} key Key to identify the param
+     * @returns {boolean} 
+     * @memberof Params
+     */
+    isUnique(key: string) {
+        return this.unique.hasOwnProperty(key);
+    }
+    /**
+     * Checks if param object has heritable attribute.
+     * @param {String} key Key to identify the param
+     * @returns {boolean}
+     * @memberof Params
+     */
+    isHeritable(key: string) {
+        return this.heritable.hasOwnProperty(key);
+    }
+    /**
+     * Checks if param has unique key.
+     * @param {Parameters} params Param to check if it is unique 
+     * @returns {boolean}
+     * @memberof Params
+     */
+    hasAllUniques(params: {[index:string]: Parameter}) {
+        //for (let key in this.unique) if (!params.hasOwnProperty(key)) return false;
+        return true;
+        throw new Error("Unique check inoperable, always returns true");
+    }
+    /**
+     * Returns the expected type for a specific param.
+     * @param {String} key Identifier of the param
+     * @param {*} expected 
+     * @param {*} actual 
+     * @memberof Params
+     * @returns {void}
+     */
+    wrongTypeError(key: string, expected: string, actual: string) {
+        return new Error("Parameter " + key + " is the wrong type. " + "Expected: " + expected + ", Actual: " + actual);
+    }
+    /**
+     * Turns the raw key:value pairs passed into a user-written Feature declaration
+    into key:Parameter pairs. This forces the checks for each Parameter type
+    to execute on the provided values, and should throw an error for mismatches.
+     * @param {*} values 
+     * @returns {Parameters} Returns a parameters object
+     * @memberof Params
+     */
+    __sanitizeValues(values: any) {
+        /*let newParams: any = {};
+        for (let key in values) {
+            let oldParam = values[key];
+            if (this.isUnique(key)) {
+                newParams[key] = Parameter.makeParam(this.unique[key], oldParam);
+            } else if (this.isHeritable(key)) {
+                if (values[key]) {
+                    newParams[key] = Parameter.makeParam(this.heritable[key], oldParam);
+                }
+            } else {
+                throw new Error(key + " does not exist in this set of ParamTypes: " + Object.keys(this.unique) + Object.keys(this.heritable));
+            }
+        }
+        this.__checkParams(newParams);
+        return newParams;
+        */
+       throw new Error("Sanitize values inoperable.");
```

Comment:
We'll need to figure out how to rewrite this in a different place. Since sanitize will be called in a couple of places, it's gonna break the code.